engines/bastion_katello/app/assets/javascripts/bastion_katello/tasks/task.factory.js

```diff
@@ -1,264 +1,112 @@
-    function (BastionResource, $timeout, $log, $q, CurrentOrganization) {
-        var bulkSearchRunning = false, searchIdGenerator = 0,
-            searchParamsById = {}, callbackById = {}, pollCount = 0, maxPollInterval = 10000;
-
-        var resource = BastionResource('/katello/api/v2/tasks/:id/:action',
-            {id: '@id', 'organization_id': CurrentOrganization}, {});
-
-        var foremanTasksResource = BastionResource('/foreman_tasks/api/tasks/:id/:action',
-            {},
+angular.module('Bastion.tasks').factory('Task', [
+    '$timeout',
+    'BastionResource',
+    'CurrentOrganization',
+    function($timeout, BastionResource, CurrentOrganization) {
+        var pendingTimeout;
+        var productId;
+
+        var resource = BastionResource(
+            '/foreman_tasks/api/tasks/',
+            {
+                'organization_id': CurrentOrganization,
+                isArray: true
+            },
             {
-                bulkSearch: {method: 'POST', isArray: true, params: { action: 'bulk_search'}}
+                update: { method: 'PUT' },
+                sync: { method: 'POST', params: { action: 'sync' } },
+                updateSyncPlan: {
+                    method: 'POST',
+                    params: { action: 'sync_plan' }
+                },
+                autocomplete: {
+                    method: 'GET',
+                    isArray: true,
+                    params: { id: 'auto_complete_search' }
+                }
             }
         );
 
-        function bulkSearchParams() {
-            var searches = [];
-            _.each(searchParamsById, function (searchParams, id) {
-                searchParams['search_id'] = id;
-                searches.push(searchParams);
-            });
-            return { searches: searches };
-        }
+        resource.cb = false;
 
-        function taskProgressbar(task) {
-            var mapping = { 'error': 'danger', 'warning': 'warning', 'default': 'success' };
-            var type = mapping[task.result] || mapping.default;
-            return { value: task.progress * 100, type: type };
-        }
+        resource.poll = function(params, cb) {
+            var queryParams = _.extend({}, params);
 
-        // add additional fields to the task model that help with
-        // usage in the UI templates
-        function normalizeTask(task) {
-            task.progressbar = taskProgressbar(task);
-            if (!task.humanized.errors) {
-                task.humanized.errors = [];
-            } else if (!_.isArray(task.humanized.errors)) {
-                task.humanized.errors = [task.humanized.errors];
+            if (angular.isFunction(cb)) {
+                resource.cb = cb;
             }
-        }
-
-        function updateProgress(periodic) {
-            if (_.keys(searchParamsById).length === 0) {
-                bulkSearchRunning = false;
+            if (params.product_id) {
+                productId = params.product_id;
```

Comment:
I don't think we should have product specific items in here since this task is shared amongst many resources.

Reply:
This has been changed to use `resource_id`.